modules/gapi/src/streaming/onevpl/engine/preproc/preproc_engine.cpp

```diff
@@ -0,0 +1,501 @@
+                                        ProcessingEngineBase::status_to_string(status));
+            throw std::runtime_error("cannot finalize VPP preprocessing operation");
+        }
+    } catch(const std::exception& ex) {
+        throw;
+    }
+    // obtain new frame is available
+    cv::gapi::wip::Data data;
+    get_frame(data);
+    preprocessed_frames_count++;
+    GAPI_LOG_DEBUG(nullptr, "processed frames count: " << preprocessed_frames_count);
+    return cv::util::get<cv::MediaFrame>(data);
+}
+
+ProcessingEngineBase::ExecutionStatus VPPPreprocEngine::process_error(mfxStatus status, session_type& sess) {
+    GAPI_LOG_DEBUG(nullptr, "status: " << mfxstatus_to_string(status));
+
+    switch (status) {
+        case MFX_ERR_NONE:
+        {
+            // prepare sync object for new surface
+            try {
+                sess.swap_surface(*this);
+                return ExecutionStatus::Continue;
+            } catch (const std::runtime_error& ex) {
+                GAPI_LOG_WARNING(nullptr, "[" << sess.session << "] error: " << ex.what());
+                return ExecutionStatus::Continue; // read more data
+            }
+        }
+        case MFX_ERR_MORE_DATA: // The function requires more bitstream at input before decoding can proceed
+            return ExecutionStatus::Processed;
+        case MFX_ERR_MORE_SURFACE:
+        {
+            // The function requires more frame surface at output before decoding can proceed.
+            // This applies to external memory allocations and should not be expected for
+            // a simple internal allocation case like this
+            try {
+                sess.swap_surface(*this);
+                return ExecutionStatus::Continue;
+            } catch (const std::runtime_error& ex) {
+                GAPI_LOG_WARNING(nullptr, "[" << sess.session << "] error: " << ex.what());
+                 return ExecutionStatus::Continue; // read more data
+            }
+            break;
+        }
+        case MFX_ERR_DEVICE_LOST:
+            // For non-CPU implementations,
+            // Cleanup if device is lost
+            GAPI_DbgAssert(false && "VPLLegacyDecodeEngine::process_error - "
+                                    "MFX_ERR_DEVICE_LOST is not processed");
+            break;
+        case MFX_WRN_DEVICE_BUSY:
+            // For non-CPU implementations,
+            // Wait a few milliseconds then try again
+            GAPI_DbgAssert(false && "VPLLegacyDecodeEngine::process_error - "
+                                    "MFX_WRN_DEVICE_BUSY is not processed");
+            break;
+        case MFX_WRN_VIDEO_PARAM_CHANGED:
+            // The decoder detected a new sequence header in the bitstream.
+            // Video parameters may have changed.
+            // In external memory allocation case, might need to reallocate the output surface
+            /*GAPI_DbgAssert(false && "VPLLegacyDecodeEngine::process_error - "
```

Comment:
Should it be uncommented?

Reply:
i don't know how to process this WRN because  it looks like doesn't affect on anything and happens spontaneously in the beginning of stream
I suppose we need to add logger here and make it falls through